utils/components.py

```diff
@@ -298,3 +307,339 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
+    import sys
+
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+
+    config.load_incluster_config()
+
+    with client.ApiClient() as api_client:
+        api_instance = client.CustomObjectsApi(api_client)
+
+        group = "kubeflow.org"
+        version = "v1"
+        plural = "pytorchjobs"
+
+        try:
+            api_response = api_instance.list_cluster_custom_object(
+                group, version, plural
+            )
+            print("""
+            ######################### INFO ###########################
+            # Kubeflow Training Operator PyTorchJob CRD is available #
+            ##########################################################
+            """)
+        except ApiException as e:
+            print("""
+            #################################################### ERROR ######################################################################
+            # Kubeflow Training Operator PyTorchJob CRD is unavailable. Ensure your OpenShift AI installation has Training Operator enabled #
+            #################################################################################################################################
+            """)
+            sys.exit(1)
+
+
+@dsl.component(base_image=RUNTIME_GENERIC_IMAGE)
+def test_oci_model(output_oci_model_uri: str, output_oci_registry_secret: str):
+    import base64
+    import json
+    import sys
+
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+
+    with open(
+        "/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r"
+    ) as namespace_path:
+        namespace = namespace_path.readline()
+    config.load_incluster_config()
+
+    if output_oci_model_uri is None:
+        print(f"""
+        ############################## INFO ##################################
+        # Parameter output_oci_model_uri not provided. Skipping this step... #
+        ######################################################################
+        """)
+        sys.exit(0)
+
+    # Extract from sdg_base_model parameter the registry name
+    registry_name = output_oci_model_uri.replace("oci://", "").split("/")[0]
+
+    with client.ApiClient() as api_client:
+        core_api = client.CoreV1Api(api_client)
+        try:
+            secret = core_api.read_namespaced_secret(sdg_oci_docker_secret, namespace)
+            print(f"Reading secret {sdg_oci_docker_secret} data...")
+            if secret.type == "kubernetes.io/dockerconfigjson":
+                # handle authentication if secret provided is kubernetes.io/dockerconfigjson
+                docker_config_json = json.loads(
+                    base64.b64decode(secret.data[".dockerconfigjson"]).decode("utf-8")
+                )
+                print(f"""
+                ######################## INFO ###########################
+                # OCI Secret has auth token present for {registry_name} #
```

Comment:
Since we aren't checking the registry name anymore, let's remove the registry name from the message since we didn't actually verify it.